cpp/src/neighbors/ivf_flat/ivf_flat_helpers.cu

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuvs/neighbors/ivf_flat_helpers.cuh>
+
+#include "../detail/ann_utils.cuh"
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/neighbors/detail/div_utils.hpp>
+#include <variant>
+
+namespace cuvs::neighbors::ivf_flat {
+namespace codepacker {
+namespace {
+
+template <typename T>
+__device__ void pack_1(const T* flat_code, T* block, uint32_t dim, uint32_t veclen, uint32_t offset)
```

Comment:
This is not correct- Please remove the device designator here. If these functions are needed internally then please make that more clear and move them out of the public APIs.

Reply:
These functions are not part of the public API. They are in an anonymous namespace and only used internally. The functions part of the public API are just the host side ones.

Reply:
What I'm asking is that we separate them into a different source file.

Reply:
These sre also template functions and so they can't be compiled. Please move these back to a cuh file (like src/neighbors/ivf_flat_helpers.cuh)